godtools/App/Share/SwiftUI Views/SearchBar/SearchBar.swift

```diff
@@ -0,0 +1,70 @@
+//
+//  SearchBar.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 8/29/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import SwiftUI
+
+// adapted from https://www.appcoda.com/swiftui-search-bar/
+
+struct SearchBar: View {
+    
+    @Binding var searchText: String
+    let localizationServices: LocalizationServices
+    
+    @State private var isEditing = false
+        
+    var body: some View {
+        HStack {
+         
+            TextField("", text: $searchText, onEditingChanged: { _ in
+                self.isEditing = true
+            }, onCommit: {
+                DispatchQueue.main.async {
+                    
+                    self.isEditing = false
+                }
+            })
+            .padding(7)
+            .padding(.horizontal, 27)
+            .background(Color.white)
+            .cornerRadius(6)
+            .overlay(
+                HStack {
+                    
+                    Image(systemName: "magnifyingglass")
+                        .foregroundColor(.gray)
+                        .frame(minWidth: 0, maxWidth: .infinity, alignment: .leading)
+                        .padding(.leading, 10)
+                    
+                    if isEditing {
+                        Button(action: {
+                            self.searchText = ""
+                            
+                        }) {
+                            Image(systemName: "multiply.circle.fill")
+                                .foregroundColor(.gray)
+                                .padding(.trailing, 10)
+                        }
+                    }
+                }
+            )
+            
+            if isEditing {
+                Button(action: {
+                    DispatchQueue.main.async {
+                        self.isEditing = false
+                        self.searchText = ""
+                    }
+                    
+                    UIApplication.shared.sendAction(#selector(UIResponder.resignFirstResponder), to: nil, from: nil, for: nil)
```

Comment:
We haven't had much experience using Keyboard with SwiftUI, too bad we have to use UIApplication this way to resignFirstResponder on the TextField.  Seems a bit wonky, but probably the only way at the moment.